WindowManager/app/src/main/java/com/example/windowmanagersample/RxActivity.kt

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.windowmanagersample
+
+import android.graphics.drawable.ColorDrawable
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.window.layout.FoldingFeature
+import androidx.window.layout.WindowInfoRepository
+import androidx.window.layout.WindowInfoRepository.Companion.windowInfoRepository
+import androidx.window.layout.WindowLayoutInfo
+import androidx.window.rxjava2.layout.windowLayoutInfoObservable
+import com.example.windowmanagersample.databinding.ActivityRxBinding
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+/** Demo activity that shows all display features and current device state on the screen. */
+class RxActivity : AppCompatActivity() {
+
+    private val stateLog: StringBuilder = StringBuilder()
+    private var disposable: Disposable? = null
+
+    private lateinit var binding: ActivityRxBinding
+    private lateinit var windowInfoRepository: WindowInfoRepository
+    private lateinit var observable: Observable<WindowLayoutInfo>
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityRxBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        windowInfoRepository = windowInfoRepository()
+
+        // Create a new observable
+        observable = windowInfoRepository.windowLayoutInfoObservable()
+
+        stateLog.clear()
+        stateLog.append(getString(R.string.state_update_log)).append("\n")
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        // Subscribe to receive WindowLayoutInfo updates
+        disposable?.let {
+            if (!it.isDisposed) it.dispose()
+        }
```

Comment:
Disposing should be idempotent (by contract), so we could simplify this a bit:

```suggestion
        disposable?.dispose()
```

Reply:
Ack